internal/services/network/subnet_resource.go

```diff
@@ -136,7 +136,7 @@ func resourceSubnet() *pluginsdk.Resource {
 											"Microsoft.Orbital/orbitalGateways",
 											"NGINX.NGINXPLUS/nginxDeployments",
 											"PaloAltoNetworks.Cloudngfw/firewalls",
-										}, false),
+										}, true),
```

Comment:
@magodo unfortunately we're no longer making fields case-insensitive since this ultimately requires a breaking change in the users config when we fix this later on - and over time we'll also be updating the existing fields to remove the incorrect case-insensitivity on these fields, by recasing these in the Read functions.

In the future, `hashicorp/go-azure-sdk` will be automatically recasing Constants during deserialization, which will do this for free - were Constants available for this field we'd be able to take advantage of this (so perhaps it's worth sending a Swagger PR to add these?) - however for the moment we'll need to instead normalize this value in the Read function so that these match.

Whilst unfortunately that means we need to take a different direction here, would you mind pulling this out into a variable for the interim and normalizing these values during the Read function too?

Reply:
@tombuildsstuff Sure, I'll do the normalization in the `Read` then. Regarding adding enums to the Swagger, do you know where can I get the full list of the service types? I failed to find the list from Azure document.

Reply:
@magodo from the Portal the following list is available:

```
Microsoft.AISupercomputer/accounts/jobs
Microsoft.AISupercomputer/accounts/models
Microsoft.AISupercomputer/accounts/npu
Microsoft.ApiManagement/service
Microsoft.Apollo/npu
Microsoft.App/environments
Microsoft.App/testClients
Microsoft.AVS/PrivateClouds
Microsoft.AzureCosmosDB/clusters
Microsoft.BareMetal/AzureHostedService
Microsoft.BareMetal/AzureVMware
Microsoft.BareMetal/CrayServers
Microsoft.Batch/batchAccounts
Microsoft.CloudTest/hostedpools
Microsoft.CloudTest/images
Microsoft.CloudTest/pools
Microsoft.Codespaces/plans
Microsoft.ContainerInstance/containerGroups
Microsoft.ContainerService/managedClusters
Microsoft.Databricks/workspaces
Microsoft.DBforMySQL/flexibleServers
Microsoft.DBforMySQL/serversv2
Microsoft.DBforPostgreSQL/flexibleServers
Microsoft.DBforPostgreSQL/serversv2
Microsoft.DBforPostgreSQL/singleServers
Microsoft.DelegatedNetwork/controller
Microsoft.DevCenter/networkConnection
Microsoft.DocumentDB/cassandraClusters
Microsoft.Fidalgo/networkSettings
Microsoft.HardwareSecurityModules/dedicatedHSMs
Microsoft.Kusto/clusters
Microsoft.LabServices/labplans
Microsoft.Logic/integrationServiceEnvironments
Microsoft.MachineLearningServices/workspaces
Microsoft.Netapp/volumes
Microsoft.Network/dnsResolvers
Microsoft.Orbital/orbitalGateways
Microsoft.PowerPlatform/enterprisePolicies
Microsoft.PowerPlatform/vnetaccesslinks
Microsoft.ServiceFabricMesh/networks
Microsoft.ServiceNetworking/trafficControllers
Microsoft.Singularity/accounts/jobs
Microsoft.Singularity/accounts/models
Microsoft.Singularity/accounts/npu
Microsoft.Sql/managedInstances
Microsoft.StoragePool/diskPools
Microsoft.StreamAnalytics/streamingJobs
Microsoft.Synapse/workspaces
Microsoft.Web/hostingEnvironments
Microsoft.Web/serverFarms
NGINX.NGINXPLUS/nginxDeployments
PaloAltoNetworks.Cloudngfw/firewalls
Qumulo.Storage/fileSystems
```

As such I'd take that as the canonical list for now?

Reply:
Comparing above list to the existing one, I got following:

```
0a1,3
>       "Microsoft.AISupercomputer/accounts/jobs",
>       "Microsoft.AISupercomputer/accounts/models",
>       "Microsoft.AISupercomputer/accounts/npu",
1a5,8
>       "Microsoft.Apollo/npu",
>       "Microsoft.App/environments",
>       "Microsoft.App/testClients",
>       "Microsoft.AVS/PrivateClouds",
2a10
>       "Microsoft.BareMetal/AzureHostedService",
5a14,17
>       "Microsoft.CloudTest/hostedpools",
>       "Microsoft.CloudTest/images",
>       "Microsoft.CloudTest/pools",
>       "Microsoft.Codespaces/plans",
13a26,29
>       "Microsoft.DelegatedNetwork/controller",
>       "Microsoft.DevCenter/networkConnection",
>       "Microsoft.DocumentDB/cassandraClusters",
>       "Microsoft.Fidalgo/networkSettings",
16d31
<       "Microsoft.Logic/integrationServiceEnvironments",
17a33
>       "Microsoft.Logic/integrationServiceEnvironments",
21c37,38
<       "Microsoft.Network/managedResolvers",
---
>       "Microsoft.Orbital/orbitalGateways",
>       "Microsoft.PowerPlatform/enterprisePolicies",
23a41,44
>       "Microsoft.ServiceNetworking/trafficControllers",
>       "Microsoft.Singularity/accounts/jobs",
>       "Microsoft.Singularity/accounts/models",
>       "Microsoft.Singularity/accounts/npu",
25d45
<       "Microsoft.Sql/servers",
31d50
<       "Microsoft.Orbital/orbitalGateways",
33a53
>       "Qumulo.Storage/fileSystems",
```

There are new enums, but also some pre-existed ones are missing in above new list. So I assume that list is not something fixed, but will vary in a breaking manner?

Reply:
Possibly? I'd suggest we send a Swagger PR with the combined list and then ask the Service Team to confirm?

Reply:
https://github.com/Azure/azure-rest-api-specs/pull/21838 opened.